File: examples/py/readme_example.py
from jelal import *

# Create an ordinal directly
ordinal = 1 * 31 + 13
# or using monthday
monthday = MonthDay(2, 13)
# function equals are available (discouraged)
assert monthday.ext_cmp(_monthday_new(2, 13)) == 0

ordinal_from_monthday = monthday.to_ordinal()
assert ordinal != ordinal_from_monthday

# Give to create a date
fixed_point = Date(1404, ordinal)
# Use methods on it, for example add days
expected_moved = Date(1404, ordinal + 11)
moved = fixed_point.add_days(11)
assert expected_moved.ext_cmp(moved) == 0

File: examples/c/readme_example.c
// compile with LD_LIBRARY_PATH set and `gcc file.c -ljelal`
// TODO make a Makefile for these
#include <jelal.h>

int main() {
    // Create an ordinal directly
    // if using CFFI:
    /* UOrdinal ordinal = 1 * MONTHDAY_MAX_DAY + 13; // IF CFFI */
    UOrdinal ordinal = 1 * MonthDay_MAX_DAY + 13; // IF CBINDGEN
    // or using monthday
    MonthDay monthday = monthday_new(2, 13);
    Ordinal ordinal_from_monthday = monthday_to_ordinal(&monthday);
    int has_error = ordinal != ordinal_from_monthday;

    // Give to create a date
    Date fixed_point = date_new(1404, ordinal);
    // Use methods on it, for example add days
    Date expected_moved = date_new(1404, ordinal + 11);
    Date moved = date_add_days(fixed_point, 11);
    has_error = has_error || (date_ext_cmp(&expected_moved, &moved) != 0);

    return has_error;
}
